stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/composer/MessageComposerController.kt

```diff
@@ -601,7 +603,15 @@ public class MessageComposerController(
                 channelType,
                 channelId,
                 message.copy(showInChannel = isInThread && alsoSendToChannel.value),
-            )
+            ).doOnResult(scope) { result ->
+                result.onSuccessSuspend { resultMessage ->
+                    chatClient.markMessageRead(
+                        channelType = channelType,
+                        channelId = channelId,
+                        messageId = resultMessage.id,
+                    ).await()
+                }
+            }
```

Comment:
@JcMinarro 👋 

I'm wondering why we do need to fire `markMessageRead` after every `sendMessage`.
Looks like something which needs to be done on backend side 🤔

Reply:
It is only fired on the case it was properly sent while the user is within the conversation, if, by some reason, the sendMessage process fails and the message is retried to be sent in background when the user is not within the conversation screen, it is not marked as read to avoid sync issues, like it was described here:
https://getstream.slack.com/archives/CRB9P6L03/p1721204370914149?thread_ts=1721150258.252929&cid=CRB9P6L03